cadence/contracts/test/EVMDeployer.cdc

```diff
@@ -1,20 +1,34 @@
 import "EVM"
 
-/// This contract is intended for testing purposes for the sake of capturing a deployed contract address while native
+/// This contract is intended for testing purposes for the sake of capturing a deployed contract addresses while native
 /// `evm.TransactionExecuted` event types are not available in Cadence testing framework. The deploying account should
 /// already be configured with a `CadenceOwnedAccount` resource in storage at `/storage/evm`.
 ///
 access(all) contract EVMDeployer {
 
-    access(all) let deployedAddress: EVM.EVMAddress
-
-    init(bytecode: String, value: UInt) {
-        let coa = self.account.storage.borrow<auth(EVM.Deploy) &EVM.CadenceOwnedAccount>(from: /storage/evm)
-            ?? panic("No COA found in storage")
-        self.deployedAddress = coa.deploy(
+    access(all) let deployedAddresses: {String: EVM.EVMAddress}
+    
+    access(all) fun deploy(name: String, bytecode: String, value: UInt) {
```

Comment:
Should this be `access(all)` or something more restrictive?

Reply:
This contract is a temporary test measure until I can get visibility into EVM events from within the testing framework. The problem is I need to know the EVM address of a deployed contract and can only discover that by saving the address somewhere upon deployment.

Wrapping in a resource would be the more secure practice here, but I made it a public method to reduce complexity in testing. Just added a contract-level comment at the top, lmk if you think I should move this to a resource.